tasks/main.yml

```diff
@@ -1,22 +1,18 @@
 ---
-# tasks file for openvpn
-- include_vars: "{{ item }}"
+- name: Include vars for OpenVPN installation
+  include_vars: "{{ item }}"
   with_first_found:
     - "../vars/{{ ansible_distribution }}{{ ansible_distribution_major_version }}.yml"
     - "../vars/{{ ansible_distribution }}.yml"
     - "../vars/{{ ansible_os_family }}.yml"
     - "../vars/empty.yml"
-  tags: [always]
 
 - name: Set facts
   import_tasks: set_facts.yml
-  tags: [always]
 
 - name: Uninstall OpenVPN
   import_tasks: uninstall.yml
   when: openvpn_uninstall
-  tags:
-    - uninstall
```

Comment:
Why remove the tags?

Reply:
They cause problems when the role is imported in a playbook. For example, when I import this role in my playbook but put it behind a tag, let's say `openvpn_install`, I don't want any tasks from the role to be executed if I don't pass that `openvpn_install` tag. However, the tasks that has `always` tag in this role will run regardless, making excluding the role impossible - this is actually how I noticed the issue, because I couldn't skip it - the tasks with the `always` tags were causing my playbook to fail.

Similarly, if I have `uninstall` tag in my playbook, it will unexpectedly trigger this task to be also executed.

The only way I found to work around that is to move the role to it's separate playbook.

Therefore I think tags shouldn't be used in reusable roles. If needed, those tasks can be put behind a flag.